From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Avnish Chouhan <avnish@linux.vnet.ibm.com>
Date: Thu, 23 Mar 2023 08:16:25 -0400
Subject: [PATCH] ieee1275 : Converting plain numbers to constants in Vec5

This patch converts the plain numbers used in Vec5 properties to
constants.

1. LPAR : Client program supports logical partitioning and
   associated hcall()s.
2. SPLPAR : Client program supports the Shared
   Processor LPAR Option.
3. CMO : Enables the Cooperative Memory Over-commitment Option.
4. MAX_CPU : Defines maximum number of CPUs supported.

Signed-off-by: Avnish Chouhan <avnish@linux.vnet.ibm.com>
---
 grub-core/kern/ieee1275/init.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/grub-core/kern/ieee1275/init.c b/grub-core/kern/ieee1275/init.c
index 3ea9b73b2a59..2516e02091cb 100644
--- a/grub-core/kern/ieee1275/init.c
+++ b/grub-core/kern/ieee1275/init.c
@@ -56,6 +56,12 @@ extern char _end[];
 grub_addr_t grub_ieee1275_original_stack;
 #endif
 
+#define LPAR     0x80
+#define SPLPAR   0x40
+#define BYTE2    (LPAR | SPLPAR)
+#define CMO      0x80
+#define MAX_CPU  256
+
 void
 grub_exit (int rc __attribute__((unused)))
 {
@@ -372,7 +378,7 @@ grub_ieee1275_ibm_cas (void)
     .vec4 = 0x0001, // set required minimum capacity % to the lowest value
     .vec5_size = 1 + sizeof(struct option_vector5) - 2,
     .vec5 = {
-	0, 192, 0, 128, 0, 0, 0, 0, 256
+        0, BYTE2, 0, CMO, 0, 0, 0, 0, MAX_CPU	
     }
   };
 
